src/frequenz/repo/config/mkdocs/mike.py

```diff
@@ -0,0 +1,250 @@
+    alias "latest" is added.
+
+    For pre-release tags it's the same but the "-pre" suffix is added to the version and
+    the aliases.
+
+    For branches, the title is "vX.Y-dev (SHA)", where X is the major version and Y is
+    the minor version of the branch. The version is "vX.Y-dev". If the branch is the
+    latest branch, the alias "latest-dev" is added.
+
+    Args:
+        repo_info: The repository information.
+
+    Returns:
+        The mike version.
+
+    Raises:
+        ValueError: If the given repository information is invalid or versions can't be
+            determined for some other reason.
+    """
+    title: str = ""
+    version: str = ""
+    aliases: list[str] = []
+
+    if repo_info.is_tag():
+        current_tag = repo_info.current_tag
+        if current_tag is None:
+            raise ValueError(
+                f"The tag {repo_info.ref_name!r} is not a valid semver version",
+            )
+        suffix = ""
+        if current_tag.prerelease is not None:
+            suffix = "-pre"
+        title = repo_info.ref_name  # vX.Y.Z
+        version = f"v{current_tag.major}.{current_tag.minor}{suffix}"  # vX.Y(-pre)?
+        if repo_info.is_tag_last_minor_for_major():
+            aliases.append(f"v{current_tag.major}{suffix}")  # vX(-pre)?
+        if repo_info.is_tag_latest():
+            aliases.append(f"latest{suffix}")  # latest(-pre)?
+        return MikeVersionInfo(title=title, version=version, aliases=aliases)
+
+    if repo_info.is_branch():
+        current_branch = repo_info.current_branch
+        if current_branch is None:
+            raise ValueError(
+                f"The branch {repo_info.ref_name!r} is not a valid branch name",
+            )
+        minor: int
+        if current_branch.minor is None:
+            next_minor = repo_info.find_next_minor_for_major_branch()
+            if next_minor is None:
+                raise ValueError(
+                    f"Could not determine the next minor version for {current_branch.name!r}",
+                )
+            minor = next_minor
+            aliases.append(f"v{current_branch.major}-dev")  # vX-dev
+        else:
+            minor = current_branch.minor
+
+        major = current_branch.major
+        title = f"v{major}.{minor}-dev ({repo_info.sha[:7]})"
+        version = f"v{major}.{minor}-dev"  # vX.Y-dev
+        if repo_info.is_branch_latest():
+            aliases.append("latest-dev")
+        return MikeVersionInfo(title=title, version=version, aliases=aliases)
+
+    raise ValueError(
+        f"Don't know how to handle '{repo_info.ref}' to make 'mike' version",
+    )
+
+
+_is_version_re = re.compile(r"^v(\d+).(\d+)(-dev|-pre)?$")
+_stable_to_semver_re = re.compile(r"^v(\d+).(\d+)$")
+_pre_to_semver_re = re.compile(r"^v(\d+).(\d+)-pre$")
+_dev_to_semver_re = re.compile(r"^v(\d+).(\d+)-dev$")
+
+
+def _to_fake_sortable_semver(version: str) -> str:
```

Comment:
Here you _did_ name the function different.

But this function and the ones below seem to be a duplication of earlier code?

Reply:
I'm not sure which functions you refer to, I don't see any duplicate functions, can you be more specific?

Reply:
I think I got confused because I saw different names & places for the function but you actually moved & renamed it